wallet/wallet-blockchain-interface/src/services/router-wallet.service.ts

```
@@ -0,0 +1,67 @@
+import { getEnv } from '../config/read-env-configs.js';
+import fetch from 'cross-fetch';
+
+export default class RouterWalletService {
+    private static routerWallet: RouterWalletService;
+
+    private constructor(private _address: string) {
+        this._address = _address;
+    }
+
+    public static async getRouterWallet(): Promise<RouterWalletService> {
+        if (!this.routerWallet) this.routerWallet = new RouterWalletService(getEnv('ROUTER_WALLET_ADDRESS'));
+        return this.routerWallet;
+    }
+
+    public async getHealth(): Promise<Response | any> {
+        try {
+            const address = this._address + '/health';
+            const res = await fetch(address);
+            return await res.json();
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    }
+
+    public async getInfo(): Promise<Response | any> {
+        try {
+            const address = this._address + '/info';
+            const res = await fetch(address);
+            return await res.json();
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    }
```

do we want to return an error, or is there some other way to pass up the response from the inner server? Or maybe we wouldn't even want to do that?

I guess #1 goal is that the useful error message from RUST should not be lost. and i guess #2 is that it also passes out the response code from RUST. But honestly i'm not sure we're using those right anyway

---

currently a successful response _from the wbi_ is always a 200, but the body is any errors that the rust might have generated. 

For example, trying to get an offer that fails will return a 200 OK from the WBI, but the body is:

```json
{
    "errors": [
        {
            "code": null,
            "message": "Wallet Error: Oracle error Error getting announcement: Oracle error error decoding response body: EOF while parsing a value at line 1 column 0"
        }
    ],
    "status": 400
}
```

I think this is the same way it worked before, that is, a status of 200, but a message with the failure.

---

If the router-wallet is unreachable, we get a 200 OK from the WBI, with the body of:

```json
{
    "message": "request to http://127.0.0.1:8085/info failed, reason: connect ECONNREFUSED 127.0.0.1:8085",
    "type": "system",
    "errno": "ECONNREFUSED",
    "code": "ECONNREFUSED"
}
```

lmk if you have an idea how to handle this